### DISCUSSION TITLE
Bug with disappearing sprites after implementing a camera that renders to an image

### DISCUSSION THREAD

**Discussion by cdsupina** [bevyengine/bevy#14006] [Q&A]
I'm working on implementing a live portrait for players using an additional 2D camera that maps to an image within a UiImage entity. I'm getting a strange interaction that you can see in the attached video where sometimes when I approach certain entities with a sprite, other entities of the same kind disappear and reappear as I move past them. The inconsistency tells me it is probably some kind of system ordering issue, but I am not familiar enough with this area of Bevy to understand exactly why this is happening. If anyone has any insight into why this may be happening, your input would be greatly appreciated.

Videos of the bug:

https://github.com/bevyengine/bevy/assets/15306815/b42c54a8-9530-4712-b292-a7e90cf6d71d

https://github.com/bevyengine/bevy/assets/15306815/da24a1ca-c5b1-4a96-97f3-836f524ea9a2

https://github.com/bevyengine/bevy/assets/15306815/3c37b2ed-f21e-40dd-9f0d-fdefdf275341

Here is the full commit where I introduced the proof of concept for this portrait and where the bug was introduced: https://github.com/thetawavegame/thetawave/commit/7bdfba630e8f4afcd1983e4b9b9d54faa38a1bae

The core parts are the 2D camera that I spawn as a child of the player:

```rust
.with_children(|parent| {
  parent.spawn_slot_1_ability(&abilities_res, &character.slot_1_ability);
  parent.spawn_slot_2_ability(&abilities_res, &character.slot_2_ability);

  parent.spawn(Camera2dBundle {
      transform: Transform::from_xyz(0.0, 0.0, game_parameters.camera_z)
          .with_scale(Vec3::new(0.35, 0.35, 1.0)),
      camera: Camera {
          order: 0,
          hdr: true,
          clear_color: ClearColorConfig::Custom(Color::BLUE.with_a(0.2)),
          target: RenderTarget::Image(player_portrait.texture.clone()),
          ..default()
      },
      tonemapping: Tonemapping::TonyMcMapface,
      ..default()
  });
});
```

And setting up the UiImage for the camera to render to:

```rust
let size = Extent3d {
    width: 100,
    height: 100,
    ..default()
};

let format = TextureFormat::bevy_default();
let data = vec![0; size.width as usize * size.height as usize * format.pixel_size()];

self.spawn(ImageBundle {
    style: Style {
        width: Val::Px(100.0),
        aspect_ratio: Some(1.0),
        ..default()
    },
    image: UiImage {
        texture: textures.add(Image {
            data,
            texture_descriptor: TextureDescriptor {
                usage: TextureUsages::TEXTURE_BINDING
                    | TextureUsages::COPY_DST
                    | TextureUsages::RENDER_ATTACHMENT,
                label: None,
                size,
                mip_level_count: 1,
                sample_count: 1,
                dimension: TextureDimension::D2,
                format: TextureFormat::bevy_default(),
                view_formats: &[],
            },
            sampler: ImageSampler::Default,
            texture_view_descriptor: None,
            ..default()
        }),
        ..default()
    },
    ..default()
})
.insert(PlayerPortraitUi);
```

**Comment by vgwidt** [comment]
You could check if Bevy is setting the ViewVisibility value to false: <https://bevy-cheatbook.github.io/fundamentals/visibility.html#viewvisibility>.

I have this issue with UI image overlays where Bevy seems to be culling them when the game camera is in certain positions, but I haven't figured out how to fix it in my case yet.

Edit: Adding NoFrustumCulling component to the images eliminates the issue in my case.
